File: codechef/MAXSC.java
import java.util.*;

public class MAXSC {
	public static int findMax(ArrayList<ArrayList<Integer>> scores) {
		int i = scores.size()-1;
		while (i > 0) {
			if (scores.get(i-1).size() == 0) return -1;
			Integer currElement = scores.get(i).get(0);
			Integer prevElement = scores.get(i-1).get(0);
			if (!((prevElement < currElement) & (prevElement != currElement))) {
				scores.get(i-1).remove(0);
				i++;
			}
			i--;
		}

		return 0;
	}

	public static long findSum(ArrayList<ArrayList<Integer>> scores) {
		long sum = 0;
		for (int i = 0 ; i < scores.size(); i++) {
			sum += scores.get(i).get(0);
		}
		return sum;
	}
	public static void main(String [] args) {
		Scanner input = new Scanner(System.in);
		int T = input.nextInt();
		while (T-- > 0) {
			int N = input.nextInt();
			ArrayList<ArrayList<Integer>> score = new ArrayList<ArrayList<Integer>>();
			for (int i = 0 ; i < N ; i++) {
				ArrayList<Integer> inner = new ArrayList<Integer>();
				for (int j = 0 ; j < N ; j++) {
					inner.add(input.nextInt());
				}
				Collections.sort(inner, Collections.reverseOrder());
				score.add(inner);
			}
			//System.out.println(score);
			int status = findMax(score);
			if (status != -1) {
				long sum = findSum(score);
				System.out.println(sum);
			}
			else System.out.println(-1);
		}
	}
}

File: codechef/C++/MAXSC.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int find_max(vector <vector <int>> score) {
	int i = score.size()-1;
	while(i > 0) {
		if (score.at(i-1).size() == 0) return -1;
		int curr_element = score.at(i).front();
		int prev_element = score.at(i-1).front();
		if (!((prev_element < curr_element) && (prev_element != curr_element))) {
			score.at(i-1).erase(score.at(i-1).begin());
			i++;
		}
		i--;
	}
	return 0;
}

int find_sum(vector <vector <int>> scores) {
	int sum = 0;
	for (int i = 0; i < scores.size(); i++) {
		sum += scores.at(i).front();
	}
	return sum;
}

int main() {
	int T;
	cin >> T;
	while(T--) {
		vector <vector <int>> score;
		int N;
		cin >> N;
		for (int i = 0 ; i < N ; i++) {
			vector <int> curr_scores;
			for (int j = 0 ; j < N ; j++) {
				int score;
				cin >> score;
				curr_scores.push_back(score);
			}
			sort(curr_scores.begin(), curr_scores.end(), greater<int>());
			score.push_back(curr_scores);
		}
		int status = find_max(score);
		if (status == -1) cout << -1 << endl;
		else {
			cout << find_sum(score) << endl;
		}
	}
}
